hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/inventory/ResourceManager.java

```diff
@@ -415,6 +416,29 @@ public int size(Resource<L> relativeTo) {
     }
 
     /**
+     * @param filter if not null, any resources whose IDs are in here will not exist in the returned collection
+     * @return a list of the resources in this resource manager masked by the given IDs
+     *         (that is, if an ID in this resource manager matches an ID found in the filter collection,
+     *         the resource whose ID it is will not be found in the returned collection).
+     *         DO NOT THINK THIS RETURNS ONLY THOSE RESOURCES WHOSE IDS MATCH THOSE IN FILTER, IT IS THE OPPOSITE.
+     */
+    public List<Resource<L>> getAllResources(Collection<ID> filter) {
```

Comment:
A better method or param name would preclude the need for this yelling comment, but I'll let it slide :)